### PR TITLE
network compatibility endpoints for API

### PR DESCRIPTION
**Pull request by baude**
add endpoints for networking compatibility with the API.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/api/handlers/compat/networks.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+		ConfigOnly:     networkCreate.ConfigOnly,
+		ConfigFrom:     networkCreate.ConfigFrom,
+		Options:        networkCreate.Options,
+		Labels:         networkCreate.Labels,
+	}
+	utils.WriteResponse(w, http.StatusOK, report)
+}
+
+func RemoveNetwork(w http.ResponseWriter, r *http.Request) {
+	//runtime := r.Context().Value("runtime").(*libpod.Runtime)
```
Comment (vrothberg) [inline pkg/api/handlers/compat/networks.go:263]: Drop comment

File: pkg/api/handlers/compat/networks.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+	utils.WriteResponse(w, http.StatusOK, reports)
+}
+
+func CreateNetwork(w http.ResponseWriter, r *http.Request) {
+	var (
+		name          string
+		networkCreate types.NetworkCreateRequest
+	)
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+	//decoder := r.Context().Value("decoder").(*schema.Decoder)
```
Comment (vrothberg) [inline pkg/api/handlers/compat/networks.go:197]: Drop comment

File: pkg/api/handlers/compat/networks.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	if err := json.NewDecoder(r.Body).Decode(&networkCreate); err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "Decode()"))
+		return
+	}
+
+	if len(networkCreate.Name) > 0 {
+		name = networkCreate.Name
+	}
+	// At present I think we should just suport the bridge driver
```
Comment (TomSweeneyRedHat) [inline pkg/api/handlers/compat/networks.go:206]: nit:  I'd drop "At present I think".

**Comment by baude** [comment]
@vrothberg wrt to tests, i want to "use" this a little with some of the compat stuff to see if it fits the bill.  then yes, at that point, tests would be applicable.

File: pkg/api/handlers/compat/networks.go [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+		utils.InternalServerError(w, err)
+		return
+	}
+	name := utils.GetName(r)
+	_, err = network.InspectNetwork(config, name)
+	if err != nil {
+		// TODO our network package does not distinguish between not finding a
+		// specific network vs not being able to read it
+		utils.InternalServerError(w, err)
+	}
```
Comment (jwhonce) [inline pkg/api/handlers/compat/networks.go:54]:
```suggestion
            return
	}
```

File: pkg/api/handlers/compat/networks.go [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+		utils.InternalServerError(w, errors.New("network create only supports the bridge driver"))
+	}
+	ncOptions := entities.NetworkCreateOptions{
+		Driver:   network.DefaultNetworkDriver,
+		Internal: networkCreate.Internal,
+	}
+	if networkCreate.IPAM != nil && networkCreate.IPAM.Config != nil {
+		if len(networkCreate.IPAM.Config) > 1 {
+			utils.InternalServerError(w, errors.New("compat network create can only support one IPAM config"))
+		}
```
Comment (jwhonce) [inline pkg/api/handlers/compat/networks.go:233]:
```suggestion
                    return
		}
```

**Pull request merged**